### PR TITLE
[netatmo] Improve channel helpers

### PR DESCRIPTION
**Pull request by clinique**
This PR decorelates channel helpers from served group types, so avoid the multiplication of channel helpers.

### CODE REVIEW

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/WeatherApi.java [hunk 1]
```diff
@@ -107,6 +107,11 @@ public Collection<NAMain> getFavoriteAndGuestStationsData() throws NetatmoExcept
         return answer != null ? answer.getElements() : List.of();
     }
 
+    public Collection<NAMain> getOwnStationsData() throws NetatmoException {
```
Comment (jlaur) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/WeatherApi.java:110]: Can you rename method `getOwnedStationData` to `getOwnStationData` for consistency (line 94)?

Comment (clinique) [reply bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/WeatherApi.java]: Improved this.

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/data/ModuleType.java [hunk 2]
```diff
@@ -76,111 +62,102 @@
...
+            new ChannelGroup(RainChannelHelper.class, MeasureClass.RAIN_QUANTITY, GROUP_RAIN)),
 
     INDOOR(FeatureArea.WEATHER, "NAModule4", WEATHER_STATION,
-            List.of(MeasureCapability.class, ChannelHelperCapability.class),
-            List.of(HumidityChannelHelper.class, TemperatureExtChannelHelper.class, AirQualityChannelHelper.class,
-                    BatteryChannelHelper.class, MeasuresChannelHelper.class, TimestampExtChannelHelper.class,
-                    SignalChannelHelper.class)),
+            Set.of(MeasureCapability.class, ChannelHelperCapability.class), ChannelGroup.SIGNAL,
+            ChannelGroup.TSTAMP_EXT, ChannelGroup.MEASURE, ChannelGroup.BATTERY, ChannelGroup.HUMIDITY,
+            ChannelGroup.TEMP_OUTSIDE_EXT, ChannelGroup.AIR_QUALITY),
```
Comment (lolodomo) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/data/ModuleType.java:118]: Rather TEMP_INSIDE_EXT ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
I am a little worried about all these changes, this is more or less a new refactoring...
@clinique : did you run all the required tests ?
@jlaur : did you also test ?

**Comment by jlaur** [comment]
> @jlaur : did you also test ?

I have not tested or had a closer look at the code yet. One comment though: Perhaps it would make sense to extract the mentioned discovery fix into a separate PR to reduce risk? Cc @clinique

**Comment by lolodomo** [comment]
> I have not tested or had a closer look at the code yet. One comment though: Perhaps it would make sense to extract the mentioned discovery fix into a separate PR to reduce risk?

Yes I agree, that is always better to separate bug fixes and code enhancements.

**Comment by clinique** [comment]
You are both completely right. I should have done this in a separate PR.

**Comment by lolodomo** [comment]
@clinique : any answer to my review comment ?

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/data/ModuleType.java (hunk 2 above)
Comment (clinique) [reply bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/data/ModuleType.java]: Yes, you're correct. Changed.

**Comment by clinique** [comment]
> I am a little worried about all these changes, this is more or less a new refactoring... @clinique : did you run all the required tests ? @jlaur : did you also test ?

@lolodomo : yes, I did run tests but need some other pairs of eye. Agree, this changes many things but it seems needed to avoid multiplication of channel helpers. Impact on code base is nearly 200 LOC less.

**Review by lolodomo — APPROVED** [review]
Code LGTM

**Comment by lolodomo** [comment]
Before merging this PR, I propose to run again few tests.

**Comment by lolodomo** [comment]
@clinique : checking in MainUI the different things, I am surprised to discover a channel with label "Température Intérieure" instead of "Température Extérieure" for my outside module.
Maybe a mismatch with groups ?
image

**Comment by lolodomo** [comment]
If I restart OH without this PR but with the binding packaged with snapshot 2915, IU got this for my outside module which is clearly different and better.
image

**Comment by lolodomo** [comment]
I think I found where is the issue. I am testing the fix and then push the change to your PR.

**Comment by lolodomo** [comment]
@clinique: fix pushed as PR to your branch

**Review by lolodomo — APPROVED** [review]

**Pull request merged**